src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -40,9 +38,7 @@
 
     private int mHTTPPort = 0;
 
-    final String TAG = "--BtCon-CLIENT-Socket";
-
-    boolean mStopped = false;
+    private boolean mStopped = false;
 
     public BtToRequestSocket(BluetoothSocket socket, BtSocketDisconnectedCallBack handler,ReadyForIncoming callback) {
         print_debug("Creating BtConnectedRequestSocket");
```

Comment:
Below you try to get mmSocket.getInputStream and getOutputStream. If they fail then an exception is thrown.

First, it is an extremely bad idea to ever catch Exception directly. This hides lots of bad things. See [here](http://www.javaworld.com/article/2073800/testing-debugging/beware-the-dangers-of-generic-exceptions.html) for why this is a bad idea. Please replace the catch with the specific Exception(s) being thrown.

Second, why are we allowing the object to continue happily on its way when two key structures can't be retrieved? The object needs to go into some bad state. I'd actually argue that the right thing to do is to just throw the exception and let Connected in BtConnectorHelper handle it.